### PR TITLE
ban ip pog

### PR DESCRIPTION
**Pull request by pyronie**
adds ip ban that hopefully works with tempbans

### CODE REVIEW

File: src/main/java/mindustry/plugin/commands/ModeratorCommands.java [hunk 1]
```diff
@@ -130,6 +126,22 @@ public void registerCommands(CommandHandler handler){
             }
         });
 
+        handler.<Context>register("unbanIP", "<IP>", "Unban the specified player by IP", (args, ctx) -> {
+            PlayerData pd = getJedisData(args[0]);
```
Comment (fuzzbuck) [inline src/main/java/mindustry/plugin/commands/ModeratorCommands.java:134]: this does not work like this anymore
use Database.getData() instead

File: src/main/java/mindustry/plugin/commands/ModeratorCommands.java [hunk 2]
```diff
@@ -130,6 +126,22 @@ public void registerCommands(CommandHandler handler){
             }
         });
 
+        handler.<Context>register("unbanIP", "<IP>", "Unban the specified player by IP", (args, ctx) -> {
```
Comment (fuzzbuck) [inline src/main/java/mindustry/plugin/commands/ModeratorCommands.java:133]: change to unbanip (lowercase)

File: src/main/java/mindustry/plugin/commands/ModeratorCommands.java [hunk 3]
```diff
@@ -130,6 +126,22 @@ public void registerCommands(CommandHandler handler){
...
+            PlayerData pd = getJedisData(args[0]);
+            if(pd!= null){
+                PlayerInfo info = netServer.admins.getInfo(args[0]);
+
+                if(netServer.admins.unbanPlayerIP(pd.lastIP)){
+                    ctx.sendEmbed(true, ":wrench: unbanned IP " + pd.lastIP + " successfully!");
+                }
+                else{
+                    ctx.sendEmbed(false, ":wrench: unbanIP failed successfully, IP unable to be found");
```
Comment (fuzzbuck) [inline src/main/java/mindustry/plugin/commands/ModeratorCommands.java:142]: ???

**Review by fuzzbuck — CHANGES_REQUESTED** [review]
you also need to update getData() function with lastIP otherwise it won't even show up

File: src/main/java/mindustry/plugin/commands/ModeratorCommands.java (hunk 2 above)
Comment (fuzzbuck) [reply src/main/java/mindustry/plugin/commands/ModeratorCommands.java:133]: (applies to banip as well)

File: src/main/java/mindustry/plugin/utils/Database.java [hunk 4]
```diff
@@ -126,6 +128,7 @@ public static void updateData(String uuid, PlayerData pd){
             query.setLong(9, pd.bannedUntil);
             query.setString(10, pd.banReason);
             query.setString(11, uuid);
+            query.setString(12, pd.lastIP);
```
Comment (fuzzbuck) [inline src/main/java/mindustry/plugin/utils/Database.java:131]: switch line 130 with line 131

File: src/main/java/mindustry/plugin/commands/ModeratorCommands.java (hunk 3 above)
Comment (pyronie) [reply src/main/java/mindustry/plugin/commands/ModeratorCommands.java:145]: it succeeded at failing :P

**Comment by Volas171** [comment]
um just wanting to leave comment here (ip bans wont work anyways)

**Pull request merged**